Gems/ONNX/Code/CMakeLists.txt

```diff
@@ -0,0 +1,249 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# Currently we are in the Code folder: ${CMAKE_CURRENT_LIST_DIR}
+# Get the platform specific folder ${pal_dir} for the current folder: ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME}
+# Note: o3de_pal_dir will take care of the details for us, as this may be a restricted platform
+#       in which case it will see if that platform is present here or in the restricted folder.
+#       i.e. It could here in our gem : Gems/ONNX/Code/Platform/<platorm_name>  or
+#            <restricted_folder>/<platform_name>/Gems/ONNX/Code
+o3de_pal_dir(pal_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
+
+# Now that we have the platform abstraction layer (PAL) folder for this folder, thats where we will find the
+# traits for this platform. Traits for a platform are defines for things like whether or not something in this gem
+# is supported by this platform.
+include(${pal_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake)
+
+find_package(Onnx REQUIRED)
+
+# The ONNX.API target declares the common interface that users of this gem should depend on in their targets
+ly_add_target(
+    NAME ONNX.API INTERFACE
+    NAMESPACE Gem
+    FILES_CMAKE
+        onnx_api_files.cmake
+        ${pal_dir}/onnx_api_files.cmake
+    INCLUDE_DIRECTORIES
+        INTERFACE
+            Include
+    BUILD_DEPENDENCIES
+        INTERFACE
+           AZ::AzCore
+           Gem::ImguiAtom.Static
+           Gem::ImGui.Static
+           Gem::ImGui.ImGuiLYUtils
+)
+
+# The ONNX.Private.Object target is an internal target
+# It should not be used outside of this Gems CMakeLists.txt
+ly_add_target(
+    NAME ONNX.Private.Object STATIC
+    NAMESPACE Gem
+    FILES_CMAKE
+        onnx_private_files.cmake
+        ${pal_dir}/onnx_private_files.cmake
+    TARGET_PROPERTIES
+        O3DE_PRIVATE_TARGET TRUE
+    INCLUDE_DIRECTORIES
+        PRIVATE
+            Include
+            Source
+            ${Onnx_INCLUDE_DIR}
+    BUILD_DEPENDENCIES
+        PUBLIC
+            AZ::AzCore
+            AZ::AzFramework
+            "${Onnx_LIBS_DIR}/onnxruntime.lib"
+            Gem::ImguiAtom.Static
+            Gem::ImGui.Static
+            Gem::ImGui.ImGuiLYUtils
+)
+
+# Here add ONNX target, it depends on the Private Object library and Public API interface
+ly_add_target(
+    NAME ONNX ${PAL_TRAIT_MONOLITHIC_DRIVEN_MODULE_TYPE}
+    NAMESPACE Gem
+    FILES_CMAKE
+        onnx_shared_files.cmake
+        ${pal_dir}/onnx_shared_files.cmake
+    INCLUDE_DIRECTORIES
+        PUBLIC
+            Include
+        PRIVATE
+            Source
+    BUILD_DEPENDENCIES
+        PUBLIC
+            Gem::ONNX.API
+            "${Onnx_LIBS_DIR}/onnxruntime.lib"
```

Comment:
The 3rdParty target shouldn't be provided this library and it needs to be in a PAL layer in order to build for all of our platforms.

Reply:
Changed